### ISSUE TITLE
Editor windows briefly flash white upon opening

### ISSUE THREAD

**Issue by FractalDiane** [godotengine/godot#58056]
### Godot version

4.0.alpha2.official

### System information

WIndows 10, Vulkan, RTX 2070 Super

### Issue description

When the editor is in multi-window mode, all editor windows, such as the settings menus, are flashing white for a few frames before they appear.

This has actually been happening in every version of Godot 4 I've tried, including ones I built from source over a year ago.

https://user-images.githubusercontent.com/3895119/153767365-1cc0e4e1-4fb5-4cad-a9d6-f508a672b57f.mp4

### Steps to reproduce

1. Make sure Godot is not in single window mode (Editor Settings->Interface->Editor->Single Window Mode).
2. Open the Project Settings or Editor Settings and see if the window is briefly white upon appearing.

**Comment by Calinou** [comment]
I'm not sure if we have any control over this, as Godot is a Win32 application and cannot force the use of dark window borders (which I believe affects the default window background color).

**Comment by TechnicalSoup** [comment]
There is a little bit of discussion about this issue over at [stackoverflow](https://stackoverflow.com/questions/69715610/how-to-initialize-the-background-color-of-win32-app-to-something-other-than-whit). A few people there have mentioned that a call to SetWindowPos immediately after the window creation seems to stop the white flash.

Feels like a bit of a workaround for strange behavior in the Microsoft API, but I'll give it a try in the coming days to see what effect it has.

**Comment by TechnicalSoup** [comment]
Update on some slight progress. From experimenting with the DisplayServer code, I didn't see any real difference using some of the workarounds mentioned. However, the problem being described there was the window incorrectly painting the background color as white before the initial paint, whereas I feel as though this problem is the delay between the window being created and the first paint/render.

Looking at the code in the Windows platform DisplayServer we're not setting the background color to anything via the window class. I tested out hard coding a color in here that matched the editor interface "base" color
`wc.hbrBackground = CreateSolidBrush(RGB(51, 59, 71));`
and the momentary flash between window creation and first painting was much less jarring, however I'm still left hunting for the cause of the delay between the window creation and the first paint.

Interesting observations are that the separate call to `ShowWindow` seem redundant as by calling the `CreateWindowExW` method with the `WS_VISIBLE` flag set in the style shows the window immediately after creation. I was having trouble debugging the window paint/rendering - but it looks like the Windows message WM_PAINT sets a flag to force a redraw on the next pass through the main loop.

Also - there seemed to be a lot of unusual windows messages queued up between the window creation and the rendering that would no doubt be causing delays. On my system I was seeing a number of power broadcast messages which was quite strange.

In short - no solution as yet, but by setting a background color we can make it less obtrusive until a solution is found.

**Comment by akien-mga** [comment]
Fixed by #71289.
